php/mailchimp.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+class MailChimpException extends Exception { };
+
+/**
+ * Class MailChimp
+ */
+
+class MailChimp {
+
+	private $api_uri;
+	private $api_user = 'api_v3';
+	private $api_key = '';
+	private $user_agent = 'API V3 - PHP Sample';
+
+	public function __construct($api_key)
+	{
+		$domain = explode('-', $api_key)[1];
+		$this->api_uri = 'https://' . $domain . '.api.mailchimp.com/3.0';
+		$this->api_key = $api_key;
+	}
+
+	private function _execute_request($method, $endpoint, Array $payload = [])
+	{
+		$endpoint = $this->api_uri.$endpoint;
+
+		$ch = curl_init();
+
+		curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json'));
+		curl_setopt($ch, CURLOPT_USERPWD, $this->api_user.':'.$this->api_key);
```

Comment:
In reference to the use of a constant rather than a class property. Change to:
```
curl_setopt($ch, CURLOPT_USERPWD, self::API_USER.':'.$this->api_key);